### ISSUE TITLE
Assembly and file version set incorrectly for tags

### ISSUE THREAD

**Issue by alexeyzimarev** [adamralph/minver#385]
I am not sure if I got something wrong but the issue appeared after upgrading to 2.2.

For regular commits, everything works. For tags, the FileVersion and AssemblyVersion are mystically set completely off.

Repro:

- Clone the repo https://github.com/restsharp/RestSharp
- Run `dotnet pack -c Release -o nuget -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg`
- Check the `nuget` directory for the package and the dll has the right versions:
```
FileVersion 106.11.7
ProductVersion106.11.7-alpha.0.1+9d71791d0d092f1e0765a203b10d29edec4664a7
AssemblyVersion 106.11.7.0
```
- Check out the 106.11.6 tag
- Run the same pack command
- Observe the dll version:
```
FileVersion 106.8.10.0
ProductVersion 106.8.10+f42ec2fc409ef044ca3ba0b88c37cd39dafb1d1a
AssemblyVersion 106.8.10.0
```

We have an issue https://github.com/restsharp/RestSharp/issues/1504 and honestly, I have no clue what's wrong. Everything in the csproj file is exactly as it used to be.

**Comment by alexeyzimarev** [comment]
Found this in the directory props
```
    <Version>106.8.10</Version>
```
Have no idea how it worked before...

**Comment by adamralph** [comment]
@alexeyzimarev it could have been something to do with [this breaking change](https://github.com/dotnet/sdk/issues/10614) in .NET SDK 3.1.300. They changed around the order of the targets, which meant properties started to get set at different times.

It also caused bug in MinVer which was [fixed in 2.3.0](https://github.com/adamralph/minver/pull/347).
